iocore/net/SSLSNIConfig.cc

```diff
@@ -170,7 +170,7 @@ SNIConfigParams::Initialize()
   if (!zret.isOK()) {
     std::stringstream errMsg;
     errMsg << zret;
-    Error("%s failed to load: %s", sni_filename, errMsg.str().c_str());
+    Emergency("%s failed to load: %s", sni_filename, errMsg.str().c_str());
```

Comment:
Is this just the initial load(on startup) of the the sni.yaml?

Reply:
Yes

Reply:
It looks like reloading the sni.yaml comes here for me...what I'm missing?
https://github.com/apache/trafficserver/blob/11f987e3a84df907582ac6ea8150b414d15fc15e/iocore/net/SSLSNIConfig.cc#L196-L201

Reply:
@rob05c did you want it to die hard on a reload as well?

Reply:
Yeah, I guess that's fine. On startup, you're not breaking any running traffic. Reload is a harder decision: do we kill the app and break running traffic in the face of a potential security/access concern?

I guess not breaking running traffic is an acceptable choice here. @alficles ^

I tried to see what other configs like ip_allow.yaml do. It throws an exception and crashes ATS. But I'm not sure if that's intentional or desirable.

I also feel like a running config changing to be malformed is at least a little less likely. It seems highly likely to me that a user might create a file wrong and never notice the log error; it seems slightly less likely that a user would create a correct file, and later malform it.